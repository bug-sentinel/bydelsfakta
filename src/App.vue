<template>
  <div id="app" class="app" tabindex="-1" :class="{ menuIsOpen, navigationIsOpen }">
    <pkt-icons-sprite />
    <the-navigation-drawer />
    <div class="app__content">
      <div v-if="showWarning" class="ok-alert-container">
        <div class="ok-alert ok-alert--warning">
          {{ $t('alert.warning') }}
        </div>
      </div>
      <the-navigation-topbar />
      <main>
        <router-view />
      </main>
      <the-footer />
    </div>
    <modal v-if="isOpen" @close="isOpen = false" />
  </div>
</template>

<script>
import { mapState, mapActions } from 'vuex';
import PktIconsSprite from './components/PktIconsSprite.vue';
import TheNavigationDrawer from './components/TheNavigationDrawer.vue';
import TheNavigationTopbar from './components/TheNavigationTopbar.vue';
import TheFooter from './components/TheFooter.vue';
import Modal from './components/Modal.vue';

export default {
  name: 'App',

  components: {
    PktIconsSprite,
    TheNavigationTopbar,
    TheNavigationDrawer,
    TheFooter,
    Modal,
  },

  data: () => ({
    isOpen: false,
    showWarning: false,
    warningMessage: '',
  }),

  metaInfo() {
    return {
      title: 'Bydelsfakta – Visualisering av statistikk om befolkning, levekår og boforhold',
    };
  },

  computed: {
    ...mapState(['menuIsOpen', 'navigationIsOpen', 'ie11']),
  },

  watch: {
    $route(to) {
      if (to.params.district !== undefined) {
        this.addDistrict({ district: to.params.district, pushRoute: false });
        document.getElementById('app').focus();
      }
    },
  },

  mounted() {
    if ('ontouchstart' in document.documentElement) {
      this.setTouchDevice(true);
    }
    if (!!window.MSInputMethodContext && !!document.documentMode) {
      this.isOpen = true;
      this.setIE11Compatibility(true);
    }
  },

  created() {
    if (import.meta.env.PROD) {
      const envs = JSON.parse(window.__GLOBAL_ENVS__);
      this.setProductionMode(envs.VITE_PRODUCTION_DATA);
      document.getElementById('bydelsfakta-globals').remove();
    } else {
      this.setProductionMode(import.meta.env.VITE_PRODUCTION_DATA);
    }

    if (this.$route.name === 'Home') {
      this.addDistrict({ district: 'alle', pushRoute: false });
    }
    if (this.$route.params.district === undefined) {
      return;
    }
    this.addDistrict({ district: this.$route.params.district, pushRoute: true });
  },

  methods: {
    ...mapActions(['addDistrict', 'setTouchDevice', 'setIE11Compatibility', 'setProductionMode']),
  },
};

// Using a class on body to determine how to style focus states
document.body.addEventListener('mousedown', () => {
  document.body.classList.add('using-mouse');
});
document.body.addEventListener('keydown', () => {
  document.body.classList.remove('using-mouse');
});
</script>

<style lang="scss">
$borderRadius: 1.5px;

@mixin boxShadow($color) {
  box-shadow: 0 1px 3px rgba($color, 0.2), 0 3px 12px rgba($color, 0.05);
}

@mixin boxColor($color, $shadowColor: #292858) {
  background: rgba($color, 0.1);
  @include boxShadow($shadowColor);

  &::after {
    background: linear-gradient($color, darken($color, 5%));
  }

  &::before {
    background: $color;
  }
}

.app:focus {
  outline: none;
}

.ok-alert-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.ok-alert {
  position: relative;
  max-width: 600px;
  margin: 1rem 1.5rem 1rem;
  padding: 1rem 2.5rem 1rem 1.5rem;
  overflow: hidden;
  font-size: 1rem;
  border-radius: $borderRadius;

  &::after {
    position: absolute;
    top: 0;
    right: none;
    bottom: 0;
    left: 0;
    display: inline-flex;
    align-items: inherit;
    justify-content: center;
    width: 0.25rem;
    height: 100%;
    padding-top: 1rem;
    padding-left: 0;
    font-size: 1.75rem;
    font-family: 'Oslo Icons', sans-serif;
    text-align: center;
    content: '';
  }

  @include boxColor(#242433);

  &--warning {
    @include boxColor(#f8c66b, darken(#f8c66b, 20%));
  }
}
</style>
