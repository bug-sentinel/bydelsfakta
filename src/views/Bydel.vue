<template>
  <div class="main-container">
    <div class="main-container__cards">
      <card-image
        v-for="item in items"
        :key="item.value"
        :id="item.value"
        :kategori="item.options.kategori"
        :tema="item.options.tema"
        :bg-image="item.options.bgImage"
        :bg-color="item.options.bgColor"
        :txt-color="item.options.txtColor"
        :link="`${$route.path}/${item.value}`"
      />
    </div>
    <div class="main-container__map">
      <h3>Bydelskart</h3>
      <leaflet-map :district="geoDistricts" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import CardImage from '../components/CardImage.vue';
import LeafletMap from '../components/LeafletVue.vue';
import subpages from '../config/subpages';
import districts from '../config/geoData/districts';

export default {
  name: 'Bydel',
  components: {
    CardImage,
    LeafletMap,
  },

  props: {
    bydel: {
      type: String,
      required: true,
    },
  },

  computed: {
    ...mapGetters(['geoDistricts']),
  },

  data() {
    return {
      items: subpages,
      chosenDistrict: {},
      districts: districts,
    };
  },
};
</script>
