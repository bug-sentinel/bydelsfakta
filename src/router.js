import Vue from 'vue';
import Router from 'vue-router';
import Home from './views/Home.vue';
import Bydel from './views/Bydel.vue';
import Alder from './views/subviews/Alder.vue';
import Boligpriser from './views/subviews/Boligpriser.vue';
import Bygningstyper from './views/subviews/Bygningstyper.vue';
import Eierform from './views/subviews/Eierform.vue';
import Flytting from './views/subviews/Flytting.vue';
import Folkemengde from './views/subviews/Folkemengde.vue';
import Husholdninger from './views/subviews/Husholdninger.vue';
import Innvandrerbefolkningen from './views/subviews/Innvandrerbefolkningen.vue';
import Kommunaleboliger from './views/subviews/Kommunaleboliger.vue';
import Levekaar from './views/subviews/Levekaar.vue';
import Trangboddhet from './views/subviews/Trangboddhet.vue';

import Graph from './components/Graph.vue';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home,
    },
    {
      path: '/graphA',
      name: 'Graph1',
      props: { settings: { url: '/mockDataA.json', template: 'a' } },
      component: Graph,
    },
    {
      path: '/graphB',
      name: 'Graph2',
      props: { settings: { url: '/mockDataA2.json', template: 'a' } },
      component: Graph,
    },
    {
      path: '/graphC',
      name: 'Graph3',
      props: { settings: { url: '/mockDataB.json', template: 'b' } },
      component: Graph,
    },
    {
      path: '/bydel/:bydel',
      name: 'Bydel',
      component: Bydel,
    },
    {
      path: '/bydel/:bydel/folkemengde',
      name: 'BydelFolkemengde',
      component: Folkemengde,
    },
    {
      path: '/bydel/:bydel/alder',
      name: 'BydelAlder',
      component: Alder,
    },
    {
      path: '/bydel/:bydel/innvandrerbefolkningen',
      name: 'BydelInnvandrerbefolkningen',
      component: Innvandrerbefolkningen,
    },
    {
      path: '/bydel/:bydel/husholdninger',
      name: 'BydelHusholdninger',
      component: Husholdninger,
    },
    {
      path: '/bydel/:bydel/trangboddhet',
      name: 'BydelTrandboddhet',
      component: Trangboddhet,
    },
    {
      path: '/bydel/:bydel/eierform',
      name: 'BydelEierform',
      component: Eierform,
    },
    {
      path: '/bydel/:bydel/kommunaleboliger',
      name: 'BydelKommunaleboliger',
      component: Kommunaleboliger,
    },
    {
      path: '/bydel/:bydel/flytting',
      name: 'BydelFlytting',
      component: Flytting,
    },
    {
      path: '/bydel/:bydel/bygningstyper',
      name: 'BydelBygningstyper',
      component: Bygningstyper,
    },
    {
      path: '/bydel/:bydel/boligpriser',
      name: 'BydelBoligpriser',
      component: Boligpriser,
    },
    {
      path: '/bydel/:bydel/levekaar',
      name: 'Bydellevekaar',
      component: Levekaar,
    },
    {
      path: '/sammenlign/:bydel*',
      name: 'Sammenlign',
      component: Bydel,
      children: [
        {
          path: '/folkemengde',
          name: 'Folkemengde',
          component: Folkemengde,
        },
        {
          path: '/alder',
          name: 'Alder',
          component: Alder,
        },
        {
          path: '/innvandrerbefolkningen',
          name: 'Innvandrerbefolkningen',
          component: Innvandrerbefolkningen,
        },
        {
          path: '/husholdninger',
          name: 'Husholdninger',
          component: Husholdninger,
        },
        {
          path: '/trangboddhet',
          name: 'Trandboddhet',
          component: Trangboddhet,
        },
        {
          path: '/eierform',
          name: 'Eierform',
          component: Eierform,
        },
        {
          path: '/kommunaleboliger',
          name: 'Kommunaleboliger',
          component: Kommunaleboliger,
        },
        {
          path: '/flytting',
          name: 'Flytting',
          component: Flytting,
        },
        {
          path: '/bygningstyper',
          name: 'Bygningstyper',
          component: Bygningstyper,
        },
        {
          path: '/boligpriser',
          name: 'Boligpriser',
          component: Boligpriser,
        },
        {
          path: '/levekaar',
          name: 'levekaar',
          component: Levekaar,
        },
      ],
    },
  ],
});
