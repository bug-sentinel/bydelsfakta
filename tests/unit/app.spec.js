import { createLocalVue, mount } from '@vue/test-utils';
import App from '../../src/App.vue';
import Bydel from '../../src/views/Bydel.vue';
import router from '../../src/router';
import clickOutside from '../../src/directives/clickOutside';

describe('App', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(router);
    localVue.directive('click-outside', clickOutside);
    wrapper = mount(App, {
      localVue,
      router,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders app-component and finds id #app', () => {
    expect(wrapper.attributes('id')).toBe('app');
  });

  test('router sends you to correct subview - bydel', () => {
    router.push('/bydel/sagene');
    expect(wrapper.find(Bydel).exists()).toBe(true);
  });

  test('renders component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
