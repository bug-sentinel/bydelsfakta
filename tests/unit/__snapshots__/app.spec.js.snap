// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div
  id="app"
>
  <aside
    id="navbar"
  >
    <img
      alt="oslo-logo"
      class="oslo__logo"
      src="[object Object]"
    />
     
    <nav
      role="navigation"
    >
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-1"
          type="checkbox"
          value="1"
        />
         
        <label
          for="checkbox-1"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/gamleoslo"
        >
          Bydel Gamle Oslo
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-2"
          type="checkbox"
          value="2"
        />
         
        <label
          for="checkbox-2"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/grunerlokka"
        >
          Bydel Grünerløkka
        </a>
      </div>
      <div
        class="navigation-link navigation-link--active"
      >
        <input
          disabled="disabled"
          id="checkbox-3"
          type="checkbox"
          value="3"
        />
         
        <label
          for="checkbox-3"
        />
         
        <a
          class="navigation-link__label router-link-exact-active router-link-active"
          href="#/bydel/sagene"
        >
          Bydel Sagene
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-4"
          type="checkbox"
          value="4"
        />
         
        <label
          for="checkbox-4"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/sthanshaugen"
        >
          Bydel St. Hanshaugen
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-5"
          type="checkbox"
          value="5"
        />
         
        <label
          for="checkbox-5"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/frogner"
        >
          Bydel Frogner
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-6"
          type="checkbox"
          value="6"
        />
         
        <label
          for="checkbox-6"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/ullern"
        >
          Bydel Ullern
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-7"
          type="checkbox"
          value="7"
        />
         
        <label
          for="checkbox-7"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/vestreaker"
        >
          Bydel Vestre Aker
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-8"
          type="checkbox"
          value="8"
        />
         
        <label
          for="checkbox-8"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/nordreaker"
        >
          Bydel Nordre Aker
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-9"
          type="checkbox"
          value="9"
        />
         
        <label
          for="checkbox-9"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/bjerke"
        >
          Bydel Bjerke
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-10"
          type="checkbox"
          value="10"
        />
         
        <label
          for="checkbox-10"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/grorud"
        >
          Bydel Grorud
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-11"
          type="checkbox"
          value="11"
        />
         
        <label
          for="checkbox-11"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/stovner"
        >
          Bydel Stovner
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-12"
          type="checkbox"
          value="12"
        />
         
        <label
          for="checkbox-12"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/alna"
        >
          Bydel Alna
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-13"
          type="checkbox"
          value="13"
        />
         
        <label
          for="checkbox-13"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/ostensjo"
        >
          Bydel Østensjø
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-14"
          type="checkbox"
          value="14"
        />
         
        <label
          for="checkbox-14"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/nordstrand"
        >
          Bydel Nordstrand
        </a>
      </div>
      <div
        class="navigation-link"
      >
        <input
          id="checkbox-15"
          type="checkbox"
          value="15"
        />
         
        <label
          for="checkbox-15"
        />
         
        <a
          class="navigation-link__label"
          href="#/bydel/sondrenordstrand"
        >
          Bydel Søndre Nordstrand
        </a>
      </div>
       
      <div
        class="navigation-link navigation-link__label-compare"
      >
        <a
          class="navigation-link__label"
          href="#/bydel/alle"
        >
          Sammenlign bydeler
        </a>
      </div>
       
      <!---->
    </nav>
  </aside>
   
  <div
    id="content"
  >
    <header
      class="oslo__navigation-topbar"
    >
      <a
        class="oslo__navigation-topbar-button router-link-active"
        href="#/"
      >
        <i
          class="material-icons oslo__topbar"
        >
          arrow_back
        </i>
         
        <h4
          class="oslo__topbar oslo__topbar-text"
        >
          Bydel Sagene
        </h4>
      </a>
       
      <div
        class="navigation-topbar"
      >
        <div
          class="navigation-topbar__select"
          id="select"
          tabindex="0"
        >
          <label
            class="label"
          >
            Velg tema
          </label>
           
          <span>
            
          </span>
           
          <i
            class="material-icons"
          >
            arrow_drop_down
          </i>
        </div>
         
        <!---->
      </div>
    </header>
     
    <main>
      <div
        class="main-container"
      >
        <div
          class="main-container__cards"
        >
          <div
            class="main-container__item"
            id="folkemengde"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/folkemengde"
              style="background-image: url(/img/folkemengde.png); background-color: rgb(255, 129, 116);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(245, 173, 165);"
              >
                
      Befolkning
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Folkemengde
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="alder"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/alder"
              style="background-image: url(/img/alder.png); background-color: rgb(255, 129, 116);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(245, 173, 165);"
              >
                
      Befolkning
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Alder
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="innvandrerbefolkningen"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/innvandrerbefolkningen"
              style="background-image: url(/img/innvandring.png); background-color: rgb(255, 129, 116);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(245, 173, 165);"
              >
                
      Befolkning
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Innvandrer­befolkningen
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="husholdninger"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/husholdninger"
              style="background-image: url(/img/husholdninger.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Husholdninger
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="trangboddhet"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/trangboddhet"
              style="background-image: url(/img/trangboddhet.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Trangboddhet
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="eierform"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/eierform"
              style="background-image: url(/img/eierform.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Eierform
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="kommunaleboliger"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/kommunaleboliger"
              style="background-image: url(/img/kommunale-boliger.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Kommunale boliger
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="flytting"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/flytting"
              style="background-image: url(/img/flytting.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Flytting
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="bygningstyper"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/bygningstyper"
              style="background-image: url(/img/bygningstyper.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Bygningstyper
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="boligpriser"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/boligpriser"
              style="background-image: url(/img/boligpriser.png); background-color: rgb(66, 248, 182);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(199, 247, 201);"
              >
                
      Bomiljø
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Boligpriser
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
          <div
            class="main-container__item"
            id="levekaar"
          >
            <a
              class="oslo__card-image"
              href="#/bydel/sagene/levekaar"
              style="background-image: url(/img/levekaar.png); background-color: rgb(248, 198, 107);"
            >
              <span
                class="oslo__category"
                style="background-color: rgb(244, 206, 138);"
              >
                
      Levekår
    
              </span>
               
              <span
                class="oslo__tema"
              >
                Levekår
              </span>
               
              <span
                class="oslo__see-more-text"
              >
                Se mer
              </span>
            </a>
          </div>
        </div>
         
        <div
          class="main-container__map"
        >
          <h3>
            Bydelskart
          </h3>
        </div>
      </div>
    </main>
  </div>
</div>
`;
